pkg/client/client_v2.go

```diff
@@ -22,19 +23,72 @@ import (
 	"github.com/elastic/elastic-agent-client/v7/pkg/utils"
 )
 
-// UnitChangedType defines types for when units are adjusted.
-type UnitChangedType int
+type (
+	// UnitChangedType defines types for when units are adjusted.
+	UnitChangedType int
+	// Trigger indicates what triggered a change
+	Trigger uint
+)
 
 const (
 	// UnitChangedAdded is when a new unit is added.
-	UnitChangedAdded UnitChangedType = 1
+	UnitChangedAdded UnitChangedType = iota // unit_added
 	// UnitChangedModified is when an existing unit is modified.
-	UnitChangedModified UnitChangedType = 2
+	UnitChangedModified // unit_modified
 	// UnitChangedRemoved is when an existing unit is removed.
-	UnitChangedRemoved UnitChangedType = 3
+	UnitChangedRemoved // unit_removed
 )
 
-// String returns string representation for the unit changed type.
+const (
+	// TriggeredNothing is Trigger zero value, nothing was triggered.
+	// It exists only for completeness and documentation purposes.
+	TriggeredNothing Trigger = 0 // nothing_triggered
+
+	// TriggeredConfigChange indicates a change in config triggered the change.
+	TriggeredConfigChange Trigger = 1 << iota // config_change_triggered
```

Comment:
can you leave a comment that event can have multiple triggers and this is a mask so nobody accidentally changes this

Reply:
Done.